internal/artifacts/module_arch.go

```diff
@@ -22,35 +22,73 @@ const (
 
 // ExecuteBuild - executes build of module from Makefile
 func ExecuteBuild(source, target string, extensions []string, moduleName, platform string, wdGetter func() (string, error)) error {
+	if moduleName == "" {
+		return errors.New(buildFailedOnEmptyModuleMsg)
+	}
+
 	logs.Logger.Infof(buildMsg, moduleName)
 	loc, err := dir.Location(source, target, dir.Dev, extensions, wdGetter)
 	if err != nil {
 		return errors.Wrapf(err, buildFailedMsg, moduleName)
 	}
-	err = buildModule(loc, loc, moduleName, platform)
+	err = buildModule(loc, loc, moduleName, platform, true)
 	if err != nil {
 		return err
 	}
 	logs.Logger.Infof(buildFinishedMsg, moduleName)
 	return nil
 }
 
-// ExecuteSoloBuild - executes build of module from stand along command
+// ExecuteSoloBuild - executes build of module from stand alone command
 func ExecuteSoloBuild(source, target string, extensions []string, moduleName string, wdGetter func() (string, error)) error {
 	logs.Logger.Infof(buildMsg, moduleName)
-	loc, err := dir.Location(source, target, dir.Dev, extensions, wdGetter)
+
+	sourceDir, err := getSoloModuleBuildAbsSource(source, wdGetter)
 	if err != nil {
 		return errors.Wrapf(err, buildFailedMsg, moduleName)
 	}
-	targetLoc := dir.ModuleLocation(loc, target)
-	err = buildModule(loc, targetLoc, moduleName, "cf")
+
+	targetDir, err := getSoloModuleBuildAbsTarget(sourceDir, target, moduleName, wdGetter)
+	if err != nil {
+		return errors.Wrapf(err, buildFailedMsg, moduleName)
+	}
+
+	loc, err := dir.Location(sourceDir, targetDir, dir.Dev, extensions, wdGetter)
+	if err != nil {
+		return errors.Wrapf(err, buildFailedMsg, moduleName)
+	}
+	targetLoc := dir.ModuleLocation(loc)
+	err = buildModule(loc, targetLoc, moduleName, "cf", false)
 	if err != nil {
 		return err
 	}
 	logs.Logger.Infof(buildFinishedMsg, moduleName)
 	return nil
 }
 
+func getSoloModuleBuildAbsSource(source string, wdGetter func() (string, error)) (string, error) {
+	if source == "" {
+		return wdGetter()
+	}
+	return filepath.Abs(source)
+}
+
+func getSoloModuleBuildAbsTarget(absSource, target, moduleName string, wdGetter func() (string, error)) (string, error) {
+	if target != "" {
+		return filepath.Abs(target)
+	}
+
+	target, err := wdGetter()
+	if err != nil {
+		return "", err
+	}
+	_, projectFoilderName := filepath.Split(absSource)
+	tmpFolderName := "." + projectFoilderName + dir.TempFolderSuffix
+
+	// default target is <current folder>/.<project folder>_mta_tmp/<module_name>
+	return filepath.Join(target, tmpFolderName, moduleName), nil
```

Comment:
Is there a test for this?

Reply:
cmd/module_test, line 102

Reply:
That test doesn't use the default (temp) target, it passes the result folder as the target

Reply:
fixed test "Sanity, no target path" in "module_arch_test"